File: Kattis/C++/Building-Pyramids/Building-Pyramids.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
    cin >> n;
    int sum =0;
    int res = 0;
    int i =1;
    
    while(true)
    {
        sum += i * i;
        if(sum <= n)
        {
            res+=1;
            i+=2;
        }
        else
        {
            break;
        }
    }
    cout<<res<<endl;
}

File: Kattis/Python/Building-Pyramids.py

def main():

    n = int(input())
    sum = 0
    res = 0
    i = 1

    while True:
        
        sum += i * i
        if sum <= n:
            res += 1
            i += 2
        else:
            break

    print(res)

if __name__ == "__main__":
    main()